utils/plot.hpp

```diff
@@ -0,0 +1,184 @@
+#ifndef PLOT_HPP
+#define PLOT_HPP
+
+#define PY_SSIZE_T_CLEAN
+#include <Python.h>
+#include <string>
+
+int scatter(const std::string fname, const std::string type,const std::string figTitle, const int figWidth = 26, const int figHeight = 7) {
+
+    PyObject *pName, *pModule, *pFunc, *pArgs, *pValue;
+
+    Py_Initialize();
+    PyRun_SimpleString("import sys");
+    PyRun_SimpleString("sys.path.append(\"../utils/\")");
+    
+    pName = PyUnicode_DecodeFSDefault("plot");
+    pModule = PyImport_Import(pName);
+    
+    pFunc = PyObject_GetAttrString(pModule, "cscatter");
+    
+    pArgs = PyTuple_New(5);
+    
+    PyObject* pFname = PyString_FromString(fname.c_str());
+    PyTuple_SetItem(pArgs, 0, pFname);
+    
+    PyObject* pType = PyString_FromString(type.c_str());
+    PyTuple_SetItem(pArgs, 1, pType);
+    
+    PyObject* pFigTitle = PyString_FromString(figTitle.c_str());
+    PyTuple_SetItem(pArgs, 2, pFigTitle);
+    
+    PyObject* pFigWidth = PyLong_FromLong(figWidth);
+    PyTuple_SetItem(pArgs, 3, pFigWidth);
+    
+    PyObject* pFigHeight = PyLong_FromLong(figHeight);
+    PyTuple_SetItem(pArgs, 4, pFigHeight);
+    
+    pValue = PyObject_CallObject(pFunc, pArgs);
+    
+    return 0;
+}
+
+int barplot(const std::string fname, const std::string x, const std::string y, const std::string figTitle, const int figWidth = 5, const int figHeight = 7) {
+
+    PyObject *pName, *pModule, *pFunc, *pArgs, *pValue;
+
+    Py_Initialize();
+    PyRun_SimpleString("import sys");
+    PyRun_SimpleString("sys.path.append(\"./\")");
+    
+    pName = PyUnicode_DecodeFSDefault("plot");
+    pModule = PyImport_Import(pName);
+    
+    pFunc = PyObject_GetAttrString(pModule, "cbarplot");
+    
+    pArgs = PyTuple_New(6);
+    
+    PyObject* pFname = PyString_FromString(fname.c_str());
+    PyTuple_SetItem(pArgs, 0, pFname);
+    
+    PyObject* pX = PyString_FromString(x.c_str());
+    PyTuple_SetItem(pArgs, 1, pX);
+    
+    PyObject* pY = PyString_FromString(y.c_str());
+    PyTuple_SetItem(pArgs, 2, pY);
+    
+    PyObject* pFigTitle = PyString_FromString(figTitle.c_str());
+    PyTuple_SetItem(pArgs, 3, pFigTitle);
+    
+    PyObject* pFigWidth = PyLong_FromLong(figWidth);
+    PyTuple_SetItem(pArgs, 4, pFigWidth);
+    
+    PyObject* pFigHeight = PyLong_FromLong(figHeight);
+    PyTuple_SetItem(pArgs, 5, pFigHeight);
+    
+    pValue = PyObject_CallObject(pFunc, pArgs);
+
+    return 0;
+}
+
+int heatmap(const std::string fname, const std::string colorMap, const std::string figTitle, const int annotation = false, const int figWidth = 12, const int figHeight = 6) {
```

Comment:
```suggestion
int heatmap(const std::string& fname, const std::string& colorMap, const std::string& figTitle, const int annotation = false, const int figWidth = 12, const int figHeight = 6)
{
```

Just to match the mlpack style and to avoid an extra copy.